emergence_lib/src/graphics/mod.rs

```diff
@@ -1,214 +1,68 @@
-    /// in the [`Tilemap`] enum
-    pub const fn z_height(&self) -> f32 {
-        *self as usize as f32
-    }
+    // mesh
+    let mesh = hexagonal_plane(&map_geometry.layout);
+    let mesh_handle = meshes.add(mesh);
 
-    /// Loads the texture for this tilemap, using its corresponding [`SpriteIndex`] implementor
-    pub fn load_texture(&self, asset_server: &AssetServer) -> TilemapTexture {
-        match self {
-            Tilemap::Terrain => TerrainSprite::load(asset_server),
-            Tilemap::Organisms => OrganismSprite::load(asset_server),
-            Tilemap::Produce => ProduceSprite::load(asset_server),
-        }
-    }
+    for (terrain_entity, tile_pos, terrain) in new_terrain.iter() {
+        let pos = map_geometry.layout.hex_to_world_pos(tile_pos.hex);
 
-    /// Spawns tilemap component associated with each variant
-    pub fn spawn(
-        &self,
-        commands: &mut Commands,
-        map_geometry: &MapGeometry,
-        asset_server: &AssetServer,
-    ) -> Entity {
-        info!("Inserting TilemapBundle for {:?}...", self);
+        let color = match terrain {
+            Terrain::Plain => Color::WHITE,
+            Terrain::High => Color::YELLOW,
+            Terrain::Rocky => Color::RED,
+        };
 
-        let texture = self.load_texture(asset_server);
-        let tile_size = self.tile_size();
-        let grid_size = self.grid_size();
+        // PERF: this is wildly inefficient and lazy. Store the handles instead!
+        let material = materials.add(color.into());
 
-        let mut entity_commands = commands.spawn(TilemapBundle {
-            grid_size,
-            map_type: MAP_TYPE,
-            size: map_geometry.size(),
-            texture,
-            tile_size,
-            transform: get_tilemap_center_transform(
-                &map_geometry.size(),
-                &grid_size,
-                &MAP_TYPE,
-                self.z_height(),
-            ),
-            ..Default::default()
+        commands.entity(terrain_entity).insert(PbrBundle {
+            mesh: mesh_handle.clone(),
+            material: material.clone(),
+            transform: Transform::from_xyz(pos.x, 0.0, pos.y),
+            ..default()
         });
-
-        match self {
-            Tilemap::Terrain => entity_commands.insert(TerrainTilemap),
-            Tilemap::Organisms => entity_commands.insert(OrganismsTilemap),
-            Tilemap::Produce => entity_commands.insert(ProduceTilemap),
-        };
-
-        entity_commands.id()
     }
 }
 
-/// Manages the mapping between [`Tilemap`]s and `bevy_ecs_tilemap` tilemaps
-#[derive(Resource, Default, Debug)]
-pub struct TilemapRegister {
-    /// A vector consisting of the tilemaps initialized with each corresponding [`Tilemap`] variant.
-    pub register: Vec<TilemapId>,
+fn populate_organisms() {}
+
+fn hexagonal_plane(hex_layout: &HexLayout) -> Mesh {
+    let mesh_info = MeshInfo::hexagonal_plane(hex_layout, Hex::ZERO);
```

Comment:
```suggestion
    let mesh_info = MeshInfo::partial_hexagonal_column(hex_layout, Hex::ZERO, 1.0);
```
this will let us make a 3d hex pillar without a bottom and height of 1.0 instead of a flat hex.